src/db.h

```diff
@@ -166,11 +166,12 @@ struct media_file_info {
 
 #define mfi_offsetof(field) offsetof(struct media_file_info, field)
 
-/* PL_SMART value must be in sync with type value in Q_PL* in db.c */
+/* PL_SPECIAL value must be in sync with type value in Q_PL* in db.c */
 enum pl_type {
   PL_PLAIN = 0,
   PL_FOLDER = 1,
-  PL_SMART = 2,
+  PL_SPECIAL = 2,
+  PL_SMART = 3,
```

Comment:
Playlists will be sent to the client according to "f.type DESC", so this way it will be smart playlists before the special ones. I guess that will work, but I would prefer keeping the same order as iTunes, which I think begins with the special ones. My suggestion is to change the sort to "f.type ASC", and then have: special (0), folder (1) and then either smart or plain. Some clients (Retune) require folder pl's to come before their children pl's. However that does require a db upgrade. What do you think?